connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -196,8 +214,10 @@ private static PluginClassLoader newPluginClassLoader(
                 pluginLoaders.put(pluginClassName, inner);
                 // TODO: once versioning is enabled this line should be moved outside this if branch
                 log.info("Added plugin '{}'", pluginClassName);
+                allAddedPlugins.put(pluginClassName, new ArrayList<>());
             }
             inner.put(plugin, loader);
+            allAddedPlugins.get(pluginClassName).add(plugin);
```

Comment:
We can use `computeIfAbsent(...)` to eliminate the prior newly-added line:
```suggestion
            allAddedPlugins.computeIfAbsent(pluginClassName, n -> new ArrayList<>()).add(plugin);
```

Reply:
reused the same pattern for pluginLoader also.